File: address-geocode-us-nodejs-examples/get_distance_to_water_rest_sdk_example.js
import { GetDistanceToWaterClient } from "../address-geocode-us-nodejs/REST/get_distance_to_water_rest.js";

async function GetDistanceToWaterClientGo(licenseKey, isLive) {
    console.log("\n--------------------------------------------------");
    console.log("Address Geocode US - GetDistanceToWater - REST SDK");
    console.log("--------------------------------------------------");

    const latitude = "34.419120";
    const longitude = "-119.703421";
    const timeoutSeconds = 15;

    console.log("\n* Input *\n");
    console.log(`Latitude       : ${latitude}`);
    console.log(`Longitude      : ${longitude}`);
    console.log(`License Key    : ${licenseKey}`);
    console.log(`Is Live        : ${isLive}`);
    console.log(`Timeout Seconds: ${timeoutSeconds}`);

    try
    {
        const response = await GetDistanceToWaterClient.invoke(latitude, longitude, licenseKey, isLive, timeoutSeconds);
        console.log("\n* Distance Info *\n");
        if (response)
        {
            console.log(`MilesToWater         : ${response.DistanceToWater}`);
            console.log(`Latitude             : ${response.Latitude}`);
            console.log(`Longitude            : ${response.Longitude}`);
            console.log(`ClosestWaterLatitude : ${response.WaterLat}`);
            console.log(`ClosestWaterLongitude: ${response.WaterLon}`);
        }
        else
        {
            console.log("No distance info found.");
        }

        if (response.Error)
        {
            console.log("\n* Error *\n");
            console.log(`Error Description: ${response.Error.Desc}`);
            console.log(`Error Number     : ${response.Error.Number}`);
            console.log(`Error Location   : ${response.Error.Location}`);
        }
    }
    catch (e)
    {
        console.log("\n* Error *\n");
        console.log(`Error Message: ${e.message}`);
    }
}

export { GetDistanceToWaterClientGo };

File: address-geocode-us-dot-net-examples/GetDistanceToWaterRestSdkExample.cs
using address_geocode_us_dot_net.REST;
using System.ComponentModel;

namespace address_geocode_us_dot_net_examples
{
    public class GetDistanceToWaterRestSdkExample
    {
        public static void Go(string LicenseKey, bool IsLive)
        {
            Console.WriteLine("\r\n--------------------------------------------------");
            Console.WriteLine("Address Geocode US - GetDistanceToWater - REST SDK");
            Console.WriteLine("--------------------------------------------------");

            GetDistanceToWaterClient.GetDistanceToWaterInput getDistanceToWaterInput = new(
                Latitude: "34.419120",
                Longitude: "-119.703421",
                LicenseKey: LicenseKey,
                IsLive: IsLive,
                TimeoutSeconds: 15
            );

            Console.WriteLine("\r\n* Input *\r\n");
            Console.WriteLine($"Latitude    : {getDistanceToWaterInput.Latitude}");
            Console.WriteLine($"Longitude   : {getDistanceToWaterInput.Longitude}");
            Console.WriteLine($"License Key : {getDistanceToWaterInput.LicenseKey}");
            Console.WriteLine($"Is Live     : {getDistanceToWaterInput.IsLive}");

            GetDistanceToWaterResponse response = GetDistanceToWaterClient.Invoke(getDistanceToWaterInput);
            if (response.Error is null)
            {
                Console.WriteLine("\r\n* Distance Info *\r\n");
                Console.WriteLine($"MilesToWater          : {response.DistanceToWater}");
                Console.WriteLine($"Latitude              : {response.Latitude}");
                Console.WriteLine($"Longitude             : {response.Longitude}");
                Console.WriteLine($"ClosestWaterLatitude  : {response.WaterLat}");
                Console.WriteLine($"ClosestWaterLongitude : {response.WaterLon}");
            }
            else
            {
                Console.WriteLine("\r\n* Error *\r\n");
                Console.WriteLine($"Error Number  : {response.Error.Number}");
                Console.WriteLine($"Error Desc    : {response.Error.Desc}");
                Console.WriteLine($"Error Location: {response.Error.Location}");
            }
        }
    }
}
